crates/rmf_site_editor/src/site/slotcar.rs

```diff
@@ -56,85 +61,102 @@ fn insert_slotcar_components(
                 if let Some(desc) = model_instances.get(parent).ok().and_then(|a| a.0) {
-                    if !robot_properties.get(desc).is_ok()
-                        && is_static.get(desc).is_ok_and(|is| !is.0 .0)
-                    {
-                        description_entity = Some(desc);
-                    }
+                    description_entity = Some(desc);
                     break;
                 }
                 target_entity = parent;
             }
 
             if let Some(desc) = description_entity {
-                let mut robot = match model_descriptions.get(desc) {
-                    Ok(ModelProperty(r)) => r.clone(),
-                    Err(_) => Robot::default(),
-                };
-
-                // NOTE(@xiyuoh) It's probably a better idea to generalize this system
-                // for each RobotPropertyKind, but the current design of ChangePlugin
-                // and Robot will result in the later system run to overwrite changes
-                // from the earlier system. For now we will process data from Battery/
-                // DifferentialDrive/MechanicalSystem/AmbientSystem in a single system.
-                if let Ok(mobility_value) = serialize_robot_property_from_kind::<
-                    Mobility,
-                    DifferentialDrive,
-                >(differential_drive.clone())
-                {
-                    robot.properties.insert(Mobility::label(), mobility_value);
-                }
+                if let Ok((opt_desc_robot, opt_desc_is_static)) = model_descriptions.get(desc) {
+                    let mut robot = Robot::default();
 
-                if let Ok(power_source_value) =
-                    serialize_robot_property_from_kind::<PowerSource, Battery>(battery.clone())
-                {
-                    robot
-                        .properties
-                        .insert(PowerSource::label(), power_source_value);
-                }
+                    if let Some(desc_robot) = opt_desc_robot {
+                        if !desc_robot.0.properties.is_empty() {
+                            // The robot properties for this model description has already
+                            // been populated, do not overwrite
+                            commands
+                                .entity(e)
+                                .remove::<DifferentialDrive>()
+                                .remove::<Battery>()
+                                .remove::<AmbientSystem>()
+                                .remove::<MechanicalSystem>();
+                            continue;
+                        }
+                        robot = desc_robot.0.clone();
+                    } else {
+                        // We have to insert this ModelProperty as the ChangePlugin does
+                        // not accept changes if the component is not already present
+                        commands.entity(desc).insert(ModelProperty(robot.clone()));
```

Comment:
I don't know if using the `ChangePlugin` would be better but this comment is not quite correct, the `ChangePlugin` [allows insertion](https://github.com/open-rmf/rmf_site/blob/0ec90fdd3b4344b7c27bb10b3514952fc74d2b1d/crates/rmf_site_editor/src/site/change_plugin.rs#L41-L44) as a (default off) behavior,